aspnetcore/blazor/components.md

```diff
@@ -974,6 +974,73 @@ Components can receive route parameters from the route template provided in the
 
 Optional parameters aren't supported, so two `@page` directives are applied in the example above. The first permits navigation to the component without a parameter. The second `@page` directive takes the `{text}` route parameter and assigns the value to the `Text` property.
 
+::: moniker range=">= aspnetcore-3.1"
+
+## Partial class support
+
+Razor components are generated as partial classes. Razor components are authored using either of the following approaches:
+
+* C# code is defined in an [@code](xref:mvc/views/razor#code) block with HTML markup and Razor code in a single file. Blazor templates define their Razor components using this approach.
+* C# code is placed in a code-behind file defined as a partial class.
+
+The following example shows the default `Counter` component with an `@code` block in an app generated from a Blazor template. HTML markup, Razor code, and C# code are in the same file:
+
+*Counter.razor*:
+
+```cshtml
+@page "/counter"
+
+<h1>Counter</h1>
+
+<p>Current count: @currentCount</p>
+
+<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
+
+@code {
+    int currentCount = 0;
+
+    void IncrementCount()
+    {
+        currentCount++;
+    }
+}
+```
+
+The `Counter` component can also be created using a code-behind file with a partial class:
+
+*Counter.razor*:
+
+```cshtml
+@page "/counter"
+
+<h1>Counter</h1>
+
+<p>Current count: @currentCount</p>
+
+<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
+```
+
+*Counter.razor.cs*:
+
+```csharp
+namespace BlazorApp.Pages
+{
+    public partial class Counter
+    {
+        int currentCount = 0;
+
+        void IncrementCount()
+        {
+            currentCount++;
+        }
+    }
+}
+```
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-3.1"
+
 ## Base class inheritance for a "code-behind" experience
```

Comment:
I think we should rename and reword this section to remove the emphasis on using inheritance for code-behind. If you want a code-behind file, you should use partial classes, but we still support defining a base class for components.

```suggestion
## Specifying a component base class
```